### PR TITLE
Support for style w/o location and single style

### LINKED ISSUE #989: Setting a single style leads to undefined behavior if incorrect time of day
I'm getting a situation where I'm building a style switcher and caching each style as a property to load later. When passing them into the NavigationViewController.init() method, sometimes the style loads, and sometimes it displays like so:

simulator screen shot - iphone x - 2018-01-08 at 17 18 35

```
class ListViewController: UITableViewController {

    let styleA = StyleA()
    let styleB = StyleB()
    let styleC = StyleC()

   ...
}
```

Loading navigation from ListViewController like so:
```swift
let viewController = NavigationViewController(for: route, styles: [self.styleA])
```

Each style is a subclass of DayStyle, and implements the init() and apply() methods:
```
class StyleBase: DayStyle {
    required init() {
        super.init()
        statusBarStyle = .lightContent
    }
    
    override func apply() {
        super.apply()
        Button.appearance().textColor = #colorLiteral(red: 0.9842069745, green: 0.9843751788, blue: 0.9841964841, alpha: 1)
    }
}
```

cc @frederoni

### PR DESCRIPTION
**Pull request by frederoni**
Fixes #989 

This change will apply a style even if the receiver doesn't have a fixed
location or if only a day _or_ night style is provided.

@bsudekum @ericrwolfe 👀

### CODE REVIEW

File: MapboxNavigation/StyleManager.swift [hunk 1]
```diff
@@ -97,7 +97,24 @@ open class StyleManager: NSObject {
...
+            // We can't calculate sunset or sunrise w/o a location so just apply all styles
+            for style in styles {
+                style.apply()
+                delegate?.styleManager?(self, didApply: style)
+            }
+            return
+        }
+        
+        // Single style usage
+        guard styles.count > 1 else {
```
Comment (JThramer) [inline MapboxNavigation/StyleManager.swift]: Is this block really needed? You could easily do the same thing below via:

```swift
for style in styles {
             if style.styleType == styleTypeForTimeOfDay || styles.count == 1 {
                 style.apply()
                 currentStyleType = style.styleType
                 delegate?.styleManager?(self, didApply: style)
             }
         }
```

File: MapboxNavigation/StyleManager.swift [hunk 2]
```diff
@@ -97,7 +97,24 @@ open class StyleManager: NSObject {
     }
     
     func applyStyle() {
-        guard let location = delegate?.locationFor(styleManager: self) else { return }
+        guard let location = delegate?.locationFor(styleManager: self) else {
+            // We can't calculate sunset or sunrise w/o a location so just apply all styles
+            for style in styles {
+                style.apply()
+                delegate?.styleManager?(self, didApply: style)
```
Comment (bsudekum) [inline MapboxNavigation/StyleManager.swift]: In the case where we cannot find a style, we're going to rapid fire go through all supplied styles and apply them? What if we just applied the first?

File: MapboxNavigation/StyleManager.swift (hunk 1 above)
Comment (bsudekum) [reply MapboxNavigation/StyleManager.swift]: Yes less code is always nice, but IMO explicit guards make things a little more clear and they are easier to debug.

File: MapboxNavigation/StyleManager.swift (hunk 2 above)
Comment (frederoni) [reply MapboxNavigation/StyleManager.swift]: I assume styles.count would mostly be 1 but I could imagine e.g. `[TopBarDayStyle(), BottomBarDayStyle(), FloatingButtonsDaysStyle()]` if one class would become unwieldy. 🤔

**Pull request merged**